packages/firebase_database/firebase_database_platform_interface/lib/src/platform_interface/event.dart

```diff
@@ -57,10 +57,9 @@ class DataSnapshotPlatform {
 
     if (dataValue is Map<Object?, Object?> && childKeys != null) {
       value = {for (final key in childKeys) key: dataValue[key]};
-    } else if (dataValue is List<Object?>) {
-      value = childKeys!
-          .map((key) => dataValue[int.parse(key! as String)])
-          .toList();
+    } else if (dataValue is List<Object?> && childKeys != null) {
+      value =
+          childKeys.map((key) => dataValue[int.parse(key! as String)]).toList();
```

Comment:
This is fine, I just thought that the keys returned in order were strings anyway? So doing `int.parse(key! as String)` seems a bit odd.

Reply:
just checked, it thinks they're `Object` type.